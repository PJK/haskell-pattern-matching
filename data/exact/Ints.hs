module Ints where

-- Missing > 0
f :: Int -> Int
f 0 = 1
f x | x < 0 = 2

-- Exhaustive
g :: Int -> Int
g 0 = 1
g x | x < 0 = 2
    | x > 0 = 3
