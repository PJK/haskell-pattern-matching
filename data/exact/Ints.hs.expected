{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "f",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "VariablePattern",
                                    "contents": "~a"
                                }
                            ],
                            "Constraints:\n  ~i == False\n  ~i == ~a < 0\n  ~e == False\n  ~e == ~a == 0\n\nSatisfiable. Model:\n  ~i = False :: Bool\n  ~a =     1 :: Integer\n  ~e = False :: Bool\n"
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "f",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ],
            [
                "g",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
